database/src/main/resources/db/migration/V1_133__detected_route_changes_table.sql

```diff
@@ -0,0 +1,91 @@
+ "created-date"               DATE,
+ FOREIGN KEY ("transit-change-date", "transit-service-id") REFERENCES "gtfs-transit-changes" ("date", "transport-service-id"));
+
+DO $$
+  DECLARE
+    r RECORD;
+    c "gtfs-route-change-info";
+    stopsu INTEGER; -- "u" suffix means uppercase
+    timesu INTEGER;
+
+  BEGIN
+    FOR r IN
+      SELECT date AS d, "transport-service-id" AS tsid, "route-changes" AS rc FROM "gtfs-transit-changes"
+      LOOP
+        IF array_length(r.rc, 1) > 0 THEN
+          FOREACH c IN ARRAY r.rc
+            LOOP
+              -- Old composite is lower-inclusive, upper-exclusive '[%,%)'. New data model is both inclusive, thus reduce upper by one.
+              stopsu := COALESCE(upper(c."trip-stop-sequence-changes"), 0);
+              IF stopsu > 0 THEN
+                stopsu := stopsu -1;
+              END IF;
+              timesu := COALESCE(upper(c."trip-stop-time-changes"), 0);
+              IF timesu > 0 THEN
+                timesu := timesu - 1;
+              END IF;
+              INSERT INTO "detected-route-change"
+              VALUES (r.d, r.tsid,
+                      c."route-short-name",
+                      COALESCE(c."route-long-name",''), -- Test/staging contains data with null, migrated does not allow that
+                      c."trip-headsign",
+                      c."change-type",
+                      c."added-trips",
+                      c."removed-trips",
+                      COALESCE(lower(c."trip-stop-sequence-changes"), 0),
+                      stopsu,
+                      COALESCE(lower(c."trip-stop-time-changes"), 0),
+                      timesu,
+                      c."current-week-date",
+                      c."different-week-date",
+                      c."change-date",
+                      COALESCE(c."route-hash-id",
+                               COALESCE(c."route-short-name",'') || '-' || COALESCE(c."route-long-name",'') || '-' || COALESCE(c."trip-headsign",'')),
+                      NULL);
+            END LOOP;
+        ELSE
+          -- raise notice '      *************** skipped id=% date=%', r.tsid, r.d;
+        END IF;
+      END LOOP;
+  END
+  $$ LANGUAGE plpgsql;
+
+ALTER TABLE "gtfs-transit-changes"
+  RENAME COLUMN "route-changes" TO "route-changes-old";
+
+ALTER TYPE "gtfs-route-change-info"
+  ADD ATTRIBUTE "trip-stop-sequence-changes-lower" INTEGER;
+
+ALTER TYPE "gtfs-route-change-info"
+  ADD ATTRIBUTE "trip-stop-sequence-changes-upper" INTEGER;
+
+ALTER TYPE "gtfs-route-change-info"
+  ADD ATTRIBUTE "trip-stop-time-changes-lower" INTEGER;
+
+ALTER TYPE "gtfs-route-change-info"
+  ADD ATTRIBUTE "trip-stop-time-changes-upper" INTEGER;
+
+ALTER TYPE "gtfs-route-change-info"
+  DROP ATTRIBUTE "trip-stop-sequence-changes";
+
+ALTER TYPE "gtfs-route-change-info"
+  DROP ATTRIBUTE "trip-stop-time-changes";
+
+-- TODO: Add an index to this table to date and service-id
```

Comment:
Any perf observations from testing with prod data without indexes? Does observed behaviour show that that indexes are needed or not needed?